### PR TITLE
Seek to IFD offset

### LINKED ISSUE #3337: Exception while reading EXIF data from JPEG
There is one weird image that causes `Image.open()` to generate exception on 32-bit Python on windows. It looks like pillow thinks that EXIF data is corrupted though other applications can read exif data normally. 

### What did you do?

Call `PIL.Image.open("file.jpg")`

### What did you expect to happen?

Call to return normally without exception or complaints

### What actually happened?

Pillow generates multiple UserWarning messages about possibly corrupt EXIF, and it also raises exception on 32-bit Python build.

### What versions of Pillow and Python are you using?

Pillow 5.2.0, Python 3.6

Here is what happens on 32-bit Windows build:

```
(venv-win-py36) PS M:\> python
Python 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:04:45) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from PIL import Image
>>> Image.open('046.JPG')
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1441814 bytes but only got 0. Skipping tag 33
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 428 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 196608 bytes but only got 0. Skipping tag 60
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 196608 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 524288 bytes but only got 0. Skipping tag 240
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1048576 bytes but only got 0. Skipping tag 13222
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 2097152 bytes but only got 0. Skipping tag 1860
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 458752 bytes but only got 0. Skipping tag 10869
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 458752 bytes but only got 0. Skipping tag 1821
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 393216 bytes but only got 0. Skipping tag 1024
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 786432 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1572864 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 33752069 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "M:\venv-win-py36\lib\site-packages\PIL\Image.py", line 2609, in open
    im = _open_core(fp, filename, prefix)
  File "M:\venv-win-py36\lib\site-packages\PIL\Image.py", line 2599, in _open_core
    im = factory(fp, filename)
  File "M:\venv-win-py36\lib\site-packages\PIL\JpegImagePlugin.py", line 780, in jpeg_factory
    im = JpegImageFile(fp, filename)
  File "M:\venv-win-py36\lib\site-packages\PIL\ImageFile.py", line 102, in __init__
    self._open()
  File "M:\venv-win-py36\lib\site-packages\PIL\JpegImagePlugin.py", line 339, in _open
    handler(self, i)
  File "M:\venv-win-py36\lib\site-packages\PIL\JpegImagePlugin.py", line 124, in APP
    exif = self._getexif()
  File "M:\venv-win-py36\lib\site-packages\PIL\JpegImagePlugin.py", line 432, in _getexif
    return _getexif(self)
  File "M:\venv-win-py36\lib\site-packages\PIL\JpegImagePlugin.py", line 467, in _getexif
    info.load(file)
  File "M:\venv-win-py36\lib\site-packages\PIL\TiffImagePlugin.py", line 738, in load
    fp.seek(offset)
OverflowError: Python int too large to convert to C ssize_t
>>> import PIL
>>> PIL.__version__
'5.2.0'
```
On 64-bit Linux build I see the same warnings, but exception is not raised:
```
(venv-3.6) @ubu-srv-vm:~$ python
Python 3.6.3 (default, Oct  3 2017, 21:45:48)
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from PIL import Image
>>> Image.open("046.JPG")
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1441814 bytes but only got 0. Skipping tag 33
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 428 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 196608 bytes but only got 0. Skipping tag 60
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 196608 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 524288 bytes but only got 0. Skipping tag 240
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1048576 bytes but only got 0. Skipping tag 13222
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 2097152 bytes but only got 0. Skipping tag 1860
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 458752 bytes but only got 0. Skipping tag 10869
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 458752 bytes but only got 0. Skipping tag 1821
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 393216 bytes but only got 0. Skipping tag 1024
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 786432 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 1572864 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 33752069 bytes but only got 0. Skipping tag 0
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 24113876360 bytes but only got 0. Skipping tag 30719
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 30889119128 bytes but only got 0. Skipping tag 62463
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 238688098 bytes but only got 0. Skipping tag 63316
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 282 bytes but only got 0. Skipping tag 1
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:747: UserWarning: Possibly corrupt EXIF data.  Expecting to read 513 bytes but only got 0. Skipping tag 1
  " Skipping tag %s" % (size, len(data), tag))
/home/user/venv-3.6/lib/python3.6/site-packages/PIL/TiffImagePlugin.py:764: UserWarning: Corrupt EXIF data.  Expecting to read 12 bytes but only got 4.
  warnings.warn(str(msg))
<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=2560x1920 at 0x7F5CEF8DF080>
>>> import PIL
>>> PIL.__version__
'5.2.0'
```

I am attaching image file but that file has actual image section stripped (the whole 0xFFDA section removed, but rest of the content is intact) as this is not my photo and I don't want to upload someone else's pictures here. With this stripped image I can still see warnings and exception on 32-bit build: jpegstripped

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #3337 

The IFD offset is described in the table on page 16 (page 10 according to the bottom of the page numbering) of https://www.exif.org/Exif2-2.PDF. This is parsed at https://github.com/python-pillow/Pillow/blob/7bf5246b93cc89cfb9d6cca78c4719a943b10585/src/PIL/TiffImagePlugin.py#L491

In `TiffImageFile`, this is used in `_open` to seek to the correct position before `load()`, through https://github.com/python-pillow/Pillow/blob/7bf5246b93cc89cfb9d6cca78c4719a943b10585/src/PIL/TiffImagePlugin.py#L972-L978 and https://github.com/python-pillow/Pillow/blob/7bf5246b93cc89cfb9d6cca78c4719a943b10585/src/PIL/TiffImagePlugin.py#L1045.

However, this is not done in `_getmp()` and `_getexif()`. This PR fixes that.

### CODE REVIEW

**Comment by radarhere** [comment]
The Travis failures do not look related. My guess is that Travis has updated it's build images, and configured them incorrectly - the 'Build image provisioning date and time' for the failing jobs is 'Mon Jan 14 09:10:34 UTC 2019', while the passing jobs are at least a month old.

**Comment by hugovk** [comment]
Aha! I saw something similar in a private work repo today, failing early for no discernible reason, also with a new build image. This should be reported to Travis.

**Comment by hugovk** [comment]
https://github.com/python-pillow/Pillow/pull/3585 has a partial fix.

File: src/PIL/JpegImagePlugin.py [hunk 1]
```diff
@@ -463,6 +463,7 @@ def _getexif(self):
     head = file.read(8)
     # process dictionary
     info = TiffImagePlugin.ImageFileDirectory_v1(head)
+    file.seek(info.next)
```
Comment (hugovk) [inline src/PIL/JpegImagePlugin.py]: Unrelated, but whilst this method is being edited, shall we rename the `file` variable so it's not shadowing the builtin?

Comment (radarhere) [reply src/PIL/JpegImagePlugin.py]: Okay, I've pushed a commit.

**Pull request merged**